torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -18,7 +17,9 @@ class CheckpointImpl(Enum):
 
 class CheckpointWrapper(torch.nn.Module):
     """
-    An nn.Module that wraps another nn.Module with checkpointing.
+    An nn.Module that wraps another nn.Module with checkpointing. Note that this
+    module is not meant to be used directly, but instead it is to be used
+    through the ``checkpoint_wrapper`` function.
```

Comment:
I guess it is too late to make this private `_CheckpointWrapper`.

Reply:
I think we can discuss this further and do in follow up work, as this API is still private / prototype.